GiniPayApiLib/Classes/Documents/Payments/PaymentProvider.swift

```diff
@@ -0,0 +1,27 @@
+//
+//  PaymentProvider.swift
+//  GiniPayApiLib
+//
+//  Created by Nadya Karaban on 15.03.21.
+//
+
+import Foundation
+
+/**
+ Struct for MinAppVersions in payment provider response
+ */
+public struct MinAppVersions: Codable {
```

Comment:
Shouldn't this struct be something less visible than public?

On Android I have two classes: 
1. The response which has the same structure as the json response, which is not visible to the client
2. An result model which is obtained from a response, but it has slightly different structure that makes more sense for client.

In the case of PaymentProvider.
Response looks like this:
```
@JsonClass(generateAdapter = true)
internal data class PaymentProviderResponse(
    @Json(name = "id") val id: String,
    @Json(name = "name") val name: String,
    @Json(name = "minAppVersion") val minAppVersion: AppVersionResponse,
)

@JsonClass(generateAdapter = true)
internal data class AppVersionResponse(
    @Json(name = "android") val android: String,
)
```

And the exposed model looks like this: 
```
/**
 * A payment provider is a Gini partner which integrated the GiniPay for Banks SDK into their mobile apps.
 */
data class PaymentProvider(
    val id: String,
    val name: String,
    /**
     * The minimal required app versions per platform
     */
    val appVersion: String,
)

internal fun PaymentProviderResponse.toPaymentProvider() = PaymentProvider(
    id, name, minAppVersion.android
)
```

Notice that in the exposed model app version is a string, instead of a structure, because it doesn't make sense for the client to know about the other platform min version.

So I guess on iOS Codable classes are responses and they should not be visible. 
Is this a relevant pattern of iOS?

Reply:
agree about visibility for  min app versions and request bodies structs. 
structs like PaymentProvider will stay public since the used in public methods for Payservice in @escaping CompletionResult<PaymentProviders>